src/AutoRest.CSharp/Common/Input/InputTypes/InputModelType.cs

```diff
@@ -19,6 +26,13 @@ internal record InputModelType(string Name, string? Namespace, string? Accessibi
         public bool IsPropertyBag { get; init; } = false;
 
         public bool IsAnonymousModel { get; init; } = false;
+        public InputModelType? BaseModel { get; private set; }
+        internal void SetBaseModel(InputModelType? baseModel, [CallerFilePath] string filepath = "", [CallerMemberName] string caller = "")
```

Comment:
Could we add some xml document here to explain the reasoning of doing this?

Reply:
added.